File: calibration.py
#  560, 150:   0,   0
#  556, 365:   0, 200
#  254, 148: 300,   0

class Calibration:
  def transform(self, x, y):
    x1 = 218.
    y1 = 365.
    x2 = 346.
    y2 = 194.

    mx = (y2 - y1) / (x2 - x1)
    bx = y1 - mx * x1

    xn = int(mx*x + bx)

    x1 = 156.
    y1 = -26.
    x2 = 292.
    y2 = 147.

    my = (y2 - y1) / (x2 - x1)
    by = y1 - my * x1

    yn = int(my*y + by)

    return [xn, yn]

File: calibration.js
// run with: node calibration.js
nudged = require('nudged')

var domain = [[0, 0], [300, 0], [0, 300], [200, 100]]
var range = [[560, 108], [240, 139], [574, 432], [354, 239]]

var t = nudged.estimate('TSR', domain, range)

console.log(t)
console.log(t.transform([0, 300]))
